order-book.js

```diff
@@ -0,0 +1,99 @@
+const reconcileOrder = (existingBook, incomingOrder) => {
```

Comment:
Good job breaking things down into smaller functions, unfortunately you also split your flow over control out to multiple functions making it much harder to follow and more prone to bugs. `reconcileOrder` should be responsible for flow of control with helper functions there to perform single tasks and return data back so reconcileOrder can decide what to do next.